app/src/main/kotlin/com/mapbox/vision/teaser/replayer/SessionsAdapter.kt

```diff
@@ -1,42 +1,180 @@
 package com.mapbox.vision.teaser.replayer
 
+import android.annotation.SuppressLint
+import android.content.Context
 import android.view.LayoutInflater
 import android.view.View
+import android.view.View.INVISIBLE
+import android.view.View.VISIBLE
 import android.view.ViewGroup
+import android.widget.ImageView
 import android.widget.TextView
 import androidx.recyclerview.widget.RecyclerView
 import com.mapbox.vision.teaser.R
+import com.mapbox.vision.teaser.replayer.SessionsAdapter.AdapterItem.CameraItem
+import com.mapbox.vision.teaser.replayer.SessionsAdapter.AdapterItem.SessionItem
 import java.io.File
+import java.text.SimpleDateFormat
+import java.util.*
 
-class SessionsAdapter(basePath: String, val clickListener: (String) -> Unit) :
-    RecyclerView.Adapter<SessionsAdapter.SessionViewHolder>() {
+class SessionsAdapter(private val context: Context,
+                      private val basePath: String,
+                      private val clickSessionListener: (String) -> Unit,
+                      private val clickCameraListener: () -> Unit,
+                      private val onActivateMultiSelectionListener: () -> Unit) : RecyclerView.Adapter<SessionsAdapter.SessionViewHolder>() {
 
-    private var sessions: MutableList<String> = mutableListOf()
+    private var items: MutableList<AdapterItem> = mutableListOf()
+    private var cameraString = ""
+    private val backgroundColorSelection: Int
+    private val backgroundColorTransparent: Int
+    var selectedItems = HashSet<String>()
```

Comment:
```suggestion
    val selectedItems = HashSet<String>()
```

Reply:
could sessions have the same names? say, if they are set manually

Reply:
No for now 
Session name is folder name in the file system
Fixed var to val